client/lib/signup/step-actions/index.js

```diff
@@ -288,8 +293,10 @@ export function createSiteWithCart( callback, dependencies, stepData, reduxStore
 		saveToLocalStorageAndProceed( state, domainItem, themeItem, newSiteParams, callback );
 		return;
 	}
-
 	const locale = getLocaleSlug();
+	const isNew7dUser = !! (
+		getCurrentUserSiteCount( state ) <= 1 && isUserRegistrationDaysWithinRange( state, null, 0, 7 )
```

Comment:
When I tested, `isNew7dUser` is getting a falsey value, so `recordNewUserSiteCreated()` is not getting fired.